builder/views.py

```diff
@@ -17,6 +20,35 @@
 from .models import DraftCodelist
 
 
+def download(request, username, codelist_slug):
+    codelist = get_object_or_404(DraftCodelist, owner=username, slug=codelist_slug)
+
+    # get codes
+    codes = list(codelist.codes.values_list("code", flat=True))
```

Comment:
This will pick up all codes, whether or not they're included or excluded.

Reply:
Can I rely on the status strings to filter these?  eg `.filter(status__contains="+")`?  or do I need to build a hierarchy from the codes?

Reply:
> Can I rely on the status strings to filter these? eg .filter(status__contains="+")?

Yes.

Reply:
Brilliant, fixed in 6012368 then 🙂